discovery-provider/plugins/verified-uploads/src/handlers/users.js

```diff
@@ -1,60 +1,64 @@
-import { dp_db } from "../db.js";
-import { slack } from "../slack.js";
-import dotenv from "dotenv";
-import axios from "axios";
+import { dp_db } from '../db.js'
+import { slack } from '../slack.js'
+import dotenv from 'dotenv'
+import axios from 'axios'
+import { getPreviousState } from './utils.js'
 
-dotenv.config();
-const { IDENTITY_URL, USERS_SLACK_CHANNEL } = process.env;
+dotenv.config()
+const { IDENTITY_URL, USERS_SLACK_CHANNEL } = process.env
 const social_handle_url = (handle) =>
-  `${IDENTITY_URL}/social_handles?handle=${handle}`;
-
-export default async ({ user_id }) => {
-  const result = await dp_db("users")
-    .select("handle", "is_verified")
-    .where("user_id", "=", user_id)
-    .whereIn("is_current", ["true", "false"])
-    .orderBy("blocknumber", "desc")
-    .limit(2)
-    .catch(console.error);
-  if (result.length == 2) {
-    const current = result[0];
-    const old = result[1];
-
-    if (current.is_verified !== old.is_verified) {
-      const is_verified = current.is_verified;
-      const handle = current.handle;
-
-      // GET https://identityservice.staging.audius.co/social_handles?handle=totallynotalec
-      const { data } = await axios
-        .get(social_handle_url(handle))
-        .catch(console.error);
-
-      const { twitterVerified, instagramVerified, tikTokVerified } = data;
-
-      let source = "unknown";
-      if (twitterVerified) {
-        source = "twitter";
-      }
-      if (instagramVerified) {
-        source = "instagram";
-      }
-      if (tikTokVerified) {
-        source = "tiktok";
-      }
-
-      const header = `User *${handle}* ${
-        is_verified ? "is now" : "is no longer"
-      } verified via ${source}!`;
-
-      const body = {
-        userId: user_id,
-        handle,
-        link: `https://audius.co/${handle}`,
-        source,
-      };
-      await slack
-        .sendMsg(USERS_SLACK_CHANNEL, header, body)
-        .catch(console.error);
+  `${IDENTITY_URL}/social_handles?handle=${handle}`
+
+// TODO: send blocknumber through pg trigger
+export default async ({ user_id, blocknumber }) => {
+  const current = await dp_db('users')
+    .select('handle', 'is_verified') // get this block number
+    .where('user_id', '=', user_id)
+    .first()
+    .catch(console.error)
+  const old = await getPreviousState({
+    table: 'users',
+    id: user_id,
+    blocknumber,
+    db: dp_db
+  })
+
+  // user create
+  if (old === undefined) return
```

Comment:
does the first user creation has `is_verified` = `true` if they signed up and were verified from the start? (probably not, but maybe they could be in the same block, and in that case we should also just check if current.is_verified, and report that

Reply:
nice, good catch!

Reply:
found a couple users from 
```
select * from users where created_at = updated_at and is_verified = true;
```